pvlib/iotools/solargis.py

```diff
@@ -0,0 +1,204 @@
+    ------
+    requests.HTTPError
+        A message from the Solargis server if the request is rejected
+
+    Notes
+    -----
+    Each XML request is limited to retrieving 31 days of data.
+
+    The variable units depends on the time frequency, e.g., the unit for
+    sub-hourly irradiance data is :math:`W/m^2`, for hourly data it is
+    :math:`Wh/m^2`, and for daily data it is :math:`kWh/m^2`.
+
+    References
+    ----------
+    .. [1] `Solargis <https://solargis.com>`_
+    .. [2] `Solargis API User Guide
+       <https://solargis.atlassian.net/wiki/spaces/public/pages/7602367/Solargis+API+User+Guide>`_
+
+    Examples
+    --------
+    >>> # Retrieve two days of irradiance data from Solargis
+    >>> data, meta = response = pvlib.iotools.get_solargis(
+    >>>     latitude=48.61259, longitude=20.827079,
+    >>>     start='2022-01-01', end='2022-01-02',
+    >>>     variables=['GHI', 'DNI'], summarization='MIN_5', api_key='demo')
+    """  # noqa: E501
+    # Use pd.to_datetime so that strings (e.g. '2021-01-01') are accepted
+    start = pd.to_datetime(start)
+    end = pd.to_datetime(end)
+
+    headers = {'Content-Type': 'application/xml'}
+
+    # Solargis recommends creating a unique site_id for each location request.
+    # The site_id does not impact the data retrieval and is used for debugging.
+    site_id = f"latitude_{latitude}_longitude_{longitude}"
+
+    request_xml = f'''<ws:dataDeliveryRequest
+    dateFrom="{start.strftime('%Y-%m-%d')}"
+    dateTo="{end.strftime('%Y-%m-%d')}"
+    xmlns="http://geomodel.eu/schema/data/request"
+    xmlns:ws="http://geomodel.eu/schema/ws/data"
+    xmlns:geo="http://geomodel.eu/schema/common/geo"
+    xmlns:pv="http://geomodel.eu/schema/common/pv"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <site id="{site_id}" name="" lat="{latitude}" lng="{longitude}">
+    </site>
+    <processing key="{' '.join(variables)}" summarization="{summarization}"
+    terrainShading="{str(terrain_shading).lower()}">
+    <timestampType>{timestamp_type.upper()}</timestampType>
+    <timeZone>{tz}</timeZone>
+    </processing>
+    </ws:dataDeliveryRequest>'''
+    response = requests.post(url + "key=" + api_key, headers=headers,
+                             data=request_xml.encode('utf8'), timeout=timeout)
+
+    if response.ok is False:
+        raise requests.HTTPError(response.json())
+
+    # Parse metadata
+    header = pd.read_xml(io.StringIO(response.text),  parser='etree')
+    meta_lines = header['metadata'].iloc[0].split('#')
+    meta_lines = [line.strip() for line in meta_lines]
+    meta = {}
+    for line in meta_lines:
+        if ':' in line:
+            key = line.split(':')[0].lower()
+            if key in METADATA_FIELDS:
+                meta[key] = ':'.join(line.split(':')[1:])
+    meta['latitude'] = float(meta['latitude'])
+    meta['longitude'] = float(meta['longitude'])
+    meta['altitude'] = float(meta.pop('elevation').replace('m a.s.l.', ''))
```

Comment:
It would be great to include the data version in the returned metadata too.  Unfortunately it can't be extracted based on colons as you've done for the other fields (it looks like `'#Output from the climate database Solargis v2.2.46`).

Reply:
I agree it would be nice, but judging it to be out of scope for now.